### ISSUE TITLE
About a strange error (prolog)!

### ISSUE THREAD

**Issue by danhnguyen0811** [Jindae/seoultech-PL-2020#2]
Hello Professor!
Recently I had made an error while practicing ProLog. When I typed the following codes:
```
length(List, N),
size is N - 1,

```
The above codes failed, then I tried this
```
length(List, N),
f is 1,
size is N - f,

```
and it worked!
I still do not know what is the difference between these codes, could you please explain what actually the error is? Thank you very much!

Edited: I did not read the guide carefully so the label of this issue is missing!

**Comment by Jindae** [comment]
I need more information to answer your question.

1. What is the error here?
Prolog program states facts and rules and completed by asking queries.
What is the query here? Usually we're using `?-` prompt to represent queries.

2. What do you mean by "worked"?
Similar to the first reason, I don't know what was your intention with this program.

In Prolog, if you start with an upper case letter, it is a variable, and start with a lower case letter, it is an atom.
In your code, `size` and `f` are atoms and `List`, `N` are variables.
Also, you should end each fact and rule with `.`, but in here you used `,` instead, which has another meaning, usually logical `AND`.

So for me, the first program can be read like this.

`N` is the length of an unknown list `List`.
`size` is an atom, and you try to put `N - 1` into an atom, which is wrong.
Also, if your program is `length(List, N), size is N-1,`, then it is incomplete.
If the last `,` was `.` and `size` was `Size`, then it would mean that 
`length` of `List` is `N` and `Size` is `N - 1`.

In other words, both of the programs in your question are wrong in my eyes.

It would be helpful to give me the whole program or describe the process you've done,
so that I can explain issues more precisely.

**Comment by danhnguyen0811** [comment]
The above codes are a part of my assignment5' solution. Since the due date of this assignment is already over, I will post the full code here. It is actually ugly.
Initially, I wrote these codes for question 3.
```
reverse(L, R) :- 
    length(L, N),
    size is N - 1, 
    length(left, size),
    length(right, 1),
    append(left, right, L),
    append(R, right, Tmp),
    reverse(left, Tmp).
```
My query was 
```
reverse([a,b,1], R).
```
It returned 'false'. I tried to debug(trace) and detected the error at "Fail: size is 3-1".
After that, I edited the codes as below:
```
reverse(L, R) :- 
    length(L, N),
    f is 1,
    size is N - f,
    length(left, size),
    length(right, 1),
    append(left, right, L),
    append(R, right, Tmp),
    reverse(left, Tmp).
`` 
and it returned the desired result!
Probably this error came out because of the way I name variables~
Thank you very much for your response!
